src/Html.php

```diff
@@ -1008,6 +1008,19 @@ public static function getProgressBar(float $percentage, ?string $label = null):
         );
     }
 
+    public static function getBadgeBlock(string $text, ?string $color = '#000000'): string
+    {
+        return TemplateRenderer::getInstance()->renderFromStringTemplate(
+            <<<TWIG
+              <div class="badge_block" style="border-color: {{ color }}"><span class="me-1" style="background: {{ color }}"></span>{{ text }}
+TWIG,
+            [
+                'text' => $text,
+                'color'      => $color,
+            ]
+        );
+    }
+
```

Comment:
Maybe this should be in `fields_macros.html.twig` and adapt `datatable.html.twig`

what do you think @cedric-anne 

(I don't like twig templates generated from string in PHP)

Reply:
This is not a form field, so it has nothing to do in `fields_macros.html.twig`. 

But, indeed, a `badge` formatter could be added to `datatable.html.twig`. The entry would be an array with a mandatory `text` value and an optional `color` value.